src/schemas/product-schema.ts

```diff
@@ -0,0 +1,22 @@
+import {Schema} from 'mongoose';
```

Comment:
you have 2 types of schema - joi (validation) and mongoose (db - which should be in DAL ). separate this folder

Reply:
done